### PR TITLE
HPCC4J-657 DFSClient: Reading indices with blobs fails

### PR DESCRIPTION
**Pull request by jpmcmu**
- Updated RecordDefinitionTranslator to maintain blob information
- Added isBlob property to FieldDef
- Added translation test for indices with blobs
- Added index read test for indices with blobs
- Updated data generation to create an index with blobs

Signed-off-by: James McMullan James.McMullan@lexisnexis.com

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [x] The change has been fully tested:
  - [ ] This change does not cause any existing JUnits to fail.
  - [ ] I have include JUnit coverage to test this change
  - [ ] I have performed system test and covered possible regressions and side effects.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] This change fixes the problem, not just the symptom

### CODE REVIEW

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/HPCCFile.java [hunk 1]
```diff
@@ -250,6 +250,10 @@ private void updateProjectedRecordDef() throws Exception
                 field.setSourceType(HpccSrcType.LITTLE_ENDIAN);
             }
 
+            // if (field.isBlob())
+            // {
+            //     field.setIsBlob(false);
+            // }
         }
```
Comment (drealeed) [inline dfsclient/src/main/java/org/hpccsystems/dfs/client/HPCCFile.java:257]: is this commented out code needed?

Comment (jpmcmu) [reply dfsclient/src/main/java/org/hpccsystems/dfs/client/HPCCFile.java:259]: Yep, I have commented it back in as part of the subsequent commit

File: dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSIndexTest.java [hunk 2]
```diff
@@ -260,6 +260,27 @@ public void biasedIntTest() throws Exception
         }
     }
 
+    @Test
```
Comment (jpmcmu) [inline dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSIndexTest.java:263]: Note: This test will fail until HPCC-32875 is resolved

**Comment by jpmcmu** [comment]
@rpastrana wrapped up testing and squashed

File: commons-hpcc/src/main/java/org/hpccsystems/commons/ecl/RecordDefinitionTranslator.java [hunk 3]
```diff
@@ -416,13 +429,21 @@ else if (srcType == HpccSrcType.UTF16LE || srcType == HpccSrcType.UTF16BE)
...
             }
             default:
             {
                 throw new Exception("Unable to generate ECL unknown field type: " + field.getFieldType().description());
             }
         }
+
+        if (field.isBlob())
+        {
+            type += " {blob}";
```
Comment (rpastrana) [inline commons-hpcc/src/main/java/org/hpccsystems/commons/ecl/RecordDefinitionTranslator.java:443]: what typically precedes the ' {blob}' literal, and do we know for sure at this point type has been correctly populated with the expected pre-fix?

**Pull request merged**